### PR TITLE
XCCDF-Results to HDF

### LINKED ISSUE #95: Convert SCC XCCDF XML to a Heimdall readable format
We have results from SCC for our RHEL7 lockdown image that is in XML format and we are trying to make it readable for Heimdall.

### PR DESCRIPTION
**Pull request by zacharylc-mitre**
* Added hyperlinks to README.md
* Added mapping tool for SCC XCCDF-Results to HDF
* Made changes to heimdall_tools.rb and cli.rb to account for xccdf_mapper

Fixes #95

Signed-off-by: zacharylc <zacharylc@mitre.org>

### CODE REVIEW

**Comment by Bialogs** [comment]
needs a rebase

**Review by Bialogs — COMMENTED** [review]
Needs a review by @ejaronne

**Comment by ejaronne** [comment]
The SCC xml sample doesn't seem to produce the check-content field, as seen in the base (non-results) xccdf of the STIG. Otherwise, mappings look correct. Is it possible to see a generic expected xccdf results xml that isn't generated by any particular client? Otherwise, I fear this has been tailored to SCC's own special XML interpretation, and therefore should be renamed "SCC-xxcdf-mapper"

Actual STIG:
image

Sample from SCC (lacks Check Text):
image

**Comment by rx294** [comment]
@zacharylc-mitre Please resolve merge conflicts on the Readme

**Comment by rx294** [comment]
@zacharylc-mitre I agree with @ejaronne that the scope should be limited to SCC xccdf type.

Possibly `scc_xccdf_mapper` is the correct title ... @aaronlippold @Bialogs please add you thoughts.

Besides `xccdf_results_mapper` breaks your current naming pattern.

File: README.md [hunk 1]
```diff
@@ -5,32 +5,31 @@
...
+4.  [**fortify_mapper**](#fortify_mapper) - commercial static code analysis tool
+5.  [**jfrog_xray_mapper**](#jfrog_xray_mapper) - package vulnerability scanner
+6.  [**nessus_mapper**](#nessus_mapper) - commercial security scanner (supports compliance and vulnerability scans from Tenable.sc and Tenable.io)
+7.  [**netsparker_mapper**](#netsparker_mapper) - web application security scanner
+8.  [**nikto_mapper**](#nikto_mapper) - open-source web server scanner
+9.  [**sarif_mapper**](#sarif_mapper) - static analysis results interchange format
+10. [**scoutsuite_mapper**](#scoutsuite_mapper) - multi-cloud security auditing tool
+11. [**snyk_mapper**](#snyk_mapper) - commercial package vulnerability scanner
+12. [**sonarqube_mapper**](#sonarqube_mapper) - open-source static code analysis tool
+13. [**xccdf_results_mapper**](#xccdf_results_mapper) - SCC Client XCCDF-Results scans
```
Comment (rx294) [inline README.md:20]: Update to `SCC Compliance Validation tool`

**Review by rx294 — CHANGES_REQUESTED** [review]
Just the naming change and associated metadata changes..

mapper code LTGM

**Comment by Bialogs** [comment]
It was originally named xccdf_mapper which I renamed to xccdf_results because the XCCDF schema and the XCCDF-results schema are different. Can we point to a SCC XCCDF-Results schema? I'm trying to call it what it is. Maybe we just call it an SCC mapper.

**Comment by ejaronne** [comment]
Eventually, @aaronlippold wanted this to translate any client that generates the xccdf_results format. This has only been tested to convert SCC output. I agree though that leading with xccdf_results_mapper will go over the heads of many non-data-schema folks who just want that "SCAP mapper", but won't see it. How about we revamp: 

The upper list should have links to the converters later in the README.

In the upper list, put in an entry called "SCC (SCAP Compliance Checker) results mapper", but link it to the xccdf_results_mapper later in the README. As the xccdf_results_mapper adapts to work with other SCAP validated tools such as OpenSCAP, place a *new* entry in the upper list, but link it to the *same* xcddf_results_mapper lower on the list.

**Comment by rx294** [comment]
Adding some notes for info

looks like `test-system` field identifies the test tool that generated xccdf

**openscap**
`
        <TestResult xmlns="http://checklists.nist.gov/xccdf/1.2" id="xccdf_org.open-scap_testresult_xccdf_mil.disa.stig_profile_MAC-1_Public" start-time="2021-06-16T16:01:29-05:00" end-time="2021-06-16T16:01:37-05:00" version="001.001" test-system="cpe:/a:redhat:openscap:1.3.4">
`

**scc**
`
      <cdf:TestResult start-time="2021-04-29T14:16:20" end-time="2021-04-29T14:24:10" id="xccdf_mil.disa.stig_testresult_scap_mil.disa.stig_comp_U_RHEL_7_V3R2_STIG_SCAP_1-2_Benchmark-xccdf.xml---xccdf_mil.disa.stig_profile_MAC-1_Classified-1" version="003.002" test-system="cpe:/a:spawar:scc:5.4">
`

`test-system` seems to be universal field

https://csrc.nist.rip/library/alt-SP800-126r3.pdf
_section: 4.5 XCCDF Results_

**Comment by rx294** [comment]
Taking back my objection re `xccdf_results_mapper`

**Comment by Bialogs** [comment]
Okay so it seems the way ahead should just update the README to be clear

**Pull request merged**